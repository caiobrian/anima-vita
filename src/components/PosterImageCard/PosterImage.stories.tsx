import { Meta, Story } from '@storybook/react'
import { PosterImageCardProps } from './PosterImageCard'
import { PosterImageCard } from '.'
import FallbackImage from 'assets/img/fallback.png'

export default {
  title: 'PosterImageCard',
  component: PosterImageCard
} as Meta

const Template: Story<PosterImageCardProps> = (args) => (
  <PosterImageCard {...args} />
)

export const Default = Template.bind({})

Default.args = {
  imageUrl: FallbackImage,
  title: 'Title',
  rating: '8.1'
}
