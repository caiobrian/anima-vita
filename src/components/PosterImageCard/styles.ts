import styled from 'styled-components'
import colors from 'styles/colors'
import utils from 'styles/utils'

type Props = {
  imageUrl: string
}

export const CardImage = styled.img<Props>`
  position: absolute;
  background-image: ${(props) => `url(${props.imageUrl})`};
  background-size: cover;
  top: -1px;
  width: 102%;
  height: 102%;
  filter: grayscale(80%) brightness(60%) blur(1px);
`

export const CardTitle = styled.h3`
  font-size: 2.4rem;
  text-transform: uppercase;
  letter-spacing: 1px;
  font-weight: 700;
  z-index: 1;
  text-align: center;
  color: ${colors.text.title};
  text-overflow: none;
  display: -webkit-box;
  -webkit-line-clamp: 1;
  line-clamp: 1;
  -webkit-box-orient: vertical;
  -webkit-text-stroke: 1px ${colors.text.darker};
`

export const Card = styled.div`
  display: flex;
  flex-direction: column;
  position: relative;
  height: 40rem;
  width: 25rem;
  justify-content: flex-end;
  align-items: center;
  overflow: hidden;
  border: 1px solid black;
  outline: none;
  border-radius: 0.8rem;
  padding: 0 1.6rem;

  &:hover ${CardImage} {
    filter: grayscale(40%) brightness(80%);
  }

  &:hover ${CardTitle} {
    animation: text-animation 200ms linear;
    display: inherit;
  }

  @keyframes text-animation {
    from {
      transform: translateX(-100px);
    }
    to {
      transform: translateX(0px);
    }
  }

  ${utils.media.laptop} {
    height: 38rem;
    width: 100%;
  }

  ${utils.media.mobile} {
    height: 34rem;
    width: 100%;
  }
`

export const CardRating = styled.p`
  height: 3.4rem;
  width: 3.4rem;
  border-radius: 9.9rem;
  background-color: ${colors.rating};
  color: ${colors.text.rating};
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 1;
  font-size: 1.6rem;
  font-weight: 700;
  position: absolute;
  top: 16px;
  right: 16px;
`
