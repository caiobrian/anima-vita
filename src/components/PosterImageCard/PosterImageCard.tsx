import { Card, CardImage, CardRating, CardTitle } from './styles'

export type PosterImageCardProps = {
  imageUrl: string
  title: string
  rating: string
}

export const PosterImageCard = ({
  imageUrl,
  title,
  rating
}: PosterImageCardProps) => {
  return (
    <Card>
      <CardImage imageUrl={imageUrl} />
      <CardTitle>{title}</CardTitle>
      <CardRating>{parseFloat(rating).toFixed(1)}</CardRating>
    </Card>
  )
}
