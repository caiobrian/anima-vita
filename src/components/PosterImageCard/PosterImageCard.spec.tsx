import { render, screen } from '@testing-library/react'
import { PosterImageCard } from './PosterImageCard'

const posterImageCardData = {
  imageUrl: 'https://test.com/test.jpg',
  title: 'Anime Title',
  rating: '8.6'
}

describe('PosterImageCard', () => {
  it('should render correctly', () => {
    const { container } = render(<PosterImageCard {...posterImageCardData} />)

    expect(container).toMatchSnapshot()
  })
})
