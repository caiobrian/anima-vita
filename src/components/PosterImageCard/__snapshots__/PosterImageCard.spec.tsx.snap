// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PosterImageCard should render correctly 1`] = `
<div>
  <div
    class="sc-dkPtRN cGRddX"
  >
    <img
      class="sc-bdvvtL cYlnhX"
    />
    <h3
      class="sc-gsDKAQ kbQVSH"
    >
      Anime Title
    </h3>
    <p
      class="sc-hKwDye dLzRuZ"
    >
      8.6
    </p>
  </div>
</div>
`;
